scipy/optimize/tests/test_lsq_linear.py

```diff
@@ -204,3 +235,31 @@ class TestTRF(BaseMixin, SparseMixin):
 class TestBVLS(BaseMixin):
     method = 'bvls'
     lsq_solvers = ['exact']
+
+
+class TestErrorChecking:
+    def test_option_lsmr_tol(self):
+        # Should work with a positive float, string equal to 'auto', or None
+        _ = lsq_linear(A, b, lsq_solver='lsmr', lsmr_tol=1e-2)
+        _ = lsq_linear(A, b, lsq_solver='lsmr', lsmr_tol='auto')
+        _ = lsq_linear(A, b, lsq_solver='lsmr', lsmr_tol=None)
+
+        # Should raise error with negative float, strings
+        # other than 'auto', and integers
+        with assert_raises(ValueError):
```

Comment:
Please use pytest for that and also match the error message like so. It prevents to inadvertently catch something else.
```python
import pytest
with pytest.raises(ValueError, match="..."):
   ...
```